src/Provider/ProviderInterface.php

```diff
@@ -11,4 +12,11 @@
      * @return float
      */
     public function getRate($fromCurrency, $toCurrency);
+
+    /**
+     * Gets a list of supported currency codes
+     *
+     * @return array
+     */
+    public function getSupportedCurrencies();
```

Comment:
how about creating a different interface like `ProvidesSupportedCurrencies` for this?

Reply:
Hello ojhaujjwal, 

I would say it's a bit of over designing but you're the captain on board captain ;)

I'll do this shortly

Reply:
Well thinking about it, putting `getSupportedCurrencies` in another interface may cause an inconsistency back to the `CurrencyConverter` class ... 

let's consider this :
```php
    public function setCacheAdapter(Cache\Adapter\CacheAdapterInterface $cacheAdapter)
    {
        $this->setCachable(true);
        $this->cacheAdapter = $cacheAdapter;

        return $this;
    } 
```

if we check `$cacheAdapter` is `Cache\Adapter\CacheAdapterInterface` here, then we can't be sure `getSupportedCurrencies` is implemented in $cacheAdapter if declaration is not included in CacheAdapterInterface ... see ?